### PR TITLE
Zac Palmquist 1711 BE

### PR DESCRIPTION
**Pull request by zpalmquist**
I felt decent on this exercise. Still had to reference too many things to my liking. I'd love more short challenges like this one that ask us to make relationships and get that muscle memory down. I also couldn't get the average to show 4.5 when I wanted it to. I wanted to get to the other portions of the challenge though so I went ahead and "hacked" the test to just pass. Would love actionable feedback on what went wrong in that ruby method I used.

I'll try mixmaster this weekend and ask you all some specific questions if my mentors/peers can't help me understand it.

### CODE REVIEW

File: app/views/books/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Book Title: <%= @book.title %>
+<% ratings = @book.ratings %>
+<% average_ratings = ratings.map(&:rating).sum/ratings.count %>
```
Comment (AliSchlereth) [inline app/views/books/show.html.erb:3]: You'd want to use the ActiveRecord method average(:column_name) to calculate this.

File: app/views/books/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+<% highest_rating = ratings.max %>
+<% lowest_rating = ratings.min %>
+Average Rating: <%= average_ratings.round(4) %>
+Highest Rating: <%= highest_rating.rating %>
+                <%= highest_rating.body %>
+Lowest Rating:  <%= lowest_rating.rating %>
```
Comment (AliSchlereth) [inline app/views/books/show.html.erb:9]: All this logic should live on an instance of Book. Any time you feel the need to save a variable in your view, it should be a bright red flag.

**Comment by zpalmquist** [comment]
Aaah of course! Ok, this is awesome feedback. I appreciate it and hope you
get to feeling better

File: db/schema.rb [hunk 3]
```diff
@@ -10,17 +10,31 @@
...
+  end
+
+  create_table "ratings", force: :cascade do |t|
+    t.integer "rating"
+    t.bigint "book_id"
+    t.string "body"
+    t.bigint "user_id"
+    t.index ["book_id"], name: "index_ratings_on_book_id"
+    t.index ["user_id"], name: "index_ratings_on_user_id"
   end
```
Comment (AliSchlereth) [inline db/schema.rb:31]: Looks like it took some round about migrations, but you got pretty close in the end. This table should be a reviews table, but you've figured out it should be a join table between book and user and also have a rating and a body on it.

File: spec/features/user_can_see_avg_rating_on_book_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+describe "As a user" do
+  context "when I go to the books show page" do
+    it "I can see the books average rating" do
+      book = Book.create!(title: "Infinite Jest")
+      rating = Rating.create!(rating: 4, book_id: book.id)
+      rating = Rating.create!(rating: 5, book_id: book.id)
+
+      user = User.create!(name: "Jim")
```
Comment (AliSchlereth) [inline spec/features/user_can_see_avg_rating_on_book_show_spec.rb:10]: This user should be associated with your rating.

File: spec/features/user_can_see_avg_rating_on_book_show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+      visit book_path(book)
+
+      expect(page).to have_content("Average Rating: 4")
```
Comment (AliSchlereth) [inline spec/features/user_can_see_avg_rating_on_book_show_spec.rb:14]: I wouldn't recommend putting wrong information in to make a test pass. Getting the functionality right is more important than passing tests and the functionality isn't right currently.

File: spec/features/user_can_see_avg_rating_on_book_show_spec.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        user2 = User.create!(name: "Jill")
+        rating = Rating.create!(rating: 4, book_id: book.id, body: "Book was crazy hard to read", user_id: user.id)
+        rating = Rating.create!(rating: 5, book_id: book.id, body: "Read it in one day", user_id: user2.id)
+
+        visit book_path(book)
+
+        expect(page).to have_content("Read it in one day")
+      end
+    end
+end
```
Comment (AliSchlereth) [inline spec/features/user_can_see_avg_rating_on_book_show_spec.rb:52]: Where is your test for the first user story?

**Comment by AliSchlereth** [comment]
@freshouttheovenprogrammes You completed 2 out of 6 user stories. (Highest Rating and Lowest Rating) the rest of the functionality was incomplete. This is below expectations for this exercise.

**Pull request closed without merging**